dll/ntdll/ldr/ldrinit.c

```diff
@@ -529,7 +531,13 @@ LdrpInitializeThread(IN PCONTEXT Context)
     if (LdrpShutdownInProgress) return;
 
     /* Allocate TLS */
-    LdrpAllocateTls();
+    LdrpAllocateTls(NtCurrentTeb());
+
+    /* Add thread to active TEB list */
+    ListHead = &LdrpActiveTebList;
+    TebEntry = RtlAllocateHeap(RtlGetProcessHeap(), HEAP_ZERO_MEMORY, sizeof(LDRP_TEB_LIST_ENTRY));
```

Comment:
Avoiding hardcoding the variable type:
```suggestion
    TebEntry = RtlAllocateHeap(RtlGetProcessHeap(), HEAP_ZERO_MEMORY, sizeof(*TebEntry));
```